config/src/integration-test/java/org/springframework/security/config/annotation/authentication/ldap/NamespaceLdapAuthenticationProviderTests.java

```diff
@@ -0,0 +1,166 @@
+import org.springframework.security.config.test.SpringTestRule;
+import org.springframework.security.core.authority.mapping.GrantedAuthoritiesMapper;
+import org.springframework.security.ldap.authentication.LdapAuthenticationProvider;
+import org.springframework.security.ldap.authentication.LdapAuthenticator;
+import org.springframework.security.ldap.authentication.PasswordComparisonAuthenticator;
+import org.springframework.security.ldap.userdetails.LdapAuthoritiesPopulator;
+import org.springframework.security.web.FilterChainProxy;
+import org.springframework.security.web.access.intercept.FilterSecurityInterceptor;
+import org.springframework.test.util.ReflectionTestUtils;
+import org.springframework.test.web.servlet.MockMvc;
+
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.List;
+
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.springframework.security.test.web.servlet.request.SecurityMockMvcRequestBuilders.formLogin;
+import static org.springframework.security.test.web.servlet.response.SecurityMockMvcResultMatchers.authenticated;
+
+public class NamespaceLdapAuthenticationProviderTests {
+
+	@Rule
+	public final SpringTestRule spring = new SpringTestRule();
+
+	@Autowired
+	private MockMvc mockMvc;
+
+	@Autowired
+	private FilterChainProxy filterChainProxy;
+
+	@Test
+	public void ldapAuthenticationProvider() throws Exception {
+		this.spring.register(LdapAuthenticationProviderConfig.class).autowire();
+
+		this.mockMvc.perform(formLogin().user("bob").password("bobspassword"))
+				.andExpect(authenticated().withUsername("bob"));
+	}
+
+	@Test
+	public void ldapAuthenticationProviderCustom() {
+		this.spring.register(CustomLdapAuthenticationProviderConfig.class).autowire();
+		LdapAuthenticationProvider provider = findAuthenticationProvider(LdapAuthenticationProvider.class);
+
+		LdapAuthoritiesPopulator authoritiesPopulator = getAuthoritiesPopulator(provider);
+		assertThat(authoritiesPopulator).hasFieldOrPropertyWithValue("groupRoleAttribute", "cn");
+		assertThat(authoritiesPopulator).hasFieldOrPropertyWithValue("groupSearchBase", "ou=groups");
+		assertThat(authoritiesPopulator).hasFieldOrPropertyWithValue("groupSearchFilter", "(member={0})");
+		assertThat(getAuthoritiesMapper(provider)).hasFieldOrPropertyWithValue("prefix", "PREFIX_");
+		LdapAuthenticator authenticator = getAuthenticator(provider);
+		assertThat(callGetUserDns(authenticator, "user")).isEqualTo(Collections.singletonList("uid=user,ou=people"));
+		assertThat(authenticator).hasFieldOrPropertyWithValue("userSearch.searchBase", "ou=users");
+		assertThat(authenticator).hasFieldOrPropertyWithValue("userSearch.searchFilter", "(uid={0})");
+	}
+
+	// SEC-2490
+	@Test
+	public void ldapAuthenticationProviderCustomLdapAuthoritiesPopulator() {
+		LdapAuthoritiesPopulator LAP = Mockito.mock(LdapAuthoritiesPopulator.class);
```

Comment:
Similar to my previous comment, rather than using a mock object here, you could use a custom `LdapAuthoritiesPopulator` and check that the correct authorities are populated.
Here is an example of a custom `LdapAuthoritiesPopulator` https://github.com/spring-projects/spring-security/blob/65f5c293163f2a34e56d344b60577f5fae2325e9/ldap/src/integration-test/java/org/springframework/security/ldap/userdetails/DefaultLdapAuthoritiesPopulatorTests.java#L160-L167